borderless/explainer.md

```diff
@@ -0,0 +1,512 @@
+
+### 4. Leveraging Window Management permission for borderless
+
+In order for the borderless mode to activate, the app needs to have Window
+Management permission granted. The permission prompt can be triggered using
+JavaScript.
+
+For the **managed** context, permission prompts can be by-passed with a policy
+set by an admin and later on edited in the app’s settings. For the **unmanaged**
+context, the Window management API permission must be granted.
+
+#### Enabling Window Management
+
+Granting the window management API permission can be done with the following
+JavaScript:
+
+```
+const screenDetails = await window.getScreenDetails();
+```
+
+That will show the permission prompt to enable Window Management API permission
+which will also enable borderless capability in case the permission is given.
+
+Later on, the state of the permission can be queried with the following script:
+
+```
+navigator.permissions.query({name:'window-placement'})
+  .then((status) => {
+    // Do what you need with the permission state.
+    console.log(status.state)
+  });
+```
+
+#### Disabling Window Management
+
+To disable borderless mode, the user can disable the Window Management
+permission in App Settings.
+
+In case the Window management permission is not provided or no longer in place,
+the app will follow the normal display mode hierarchy based on the
+`display_override` and display values on the `manifest.json`. In case no other
+display mode is provided, it will
+[default to standalone mode](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/web_applications/web_app_proto_utils.cc;drc=4a8573cb240df29b0e4d9820303538fb28e31d84;l=161)
+similarly to WCO and other `display_override` modes.
+
+### 5. Isolated Web App check
+
+Borderless mode is only available for
+[Isolated Web Apps (IWAs)](https://github.com/WICG/isolated-web-apps).
+
+## Related features
+
+- [app-details feature](crbug.com/1225871) (Googlers-only)
+- [go/cros-privacy-indicators-design](go/cros-privacy-indicators-design)
+  (Googlers-only)
+- [additional windowing controls](https://github.com/ivansandrk/additional-windowing-controls/blob/main/awc-explainer.md)
+
+### Displaying app’s origin in the app settings UI
+
+Since without the title bar and the 3-dot menu, the app’s origin won’t be visible
+anymore, it needs to be shown somewhere, in this case in App Settings. This
+work has been completed as part of the [app-details feature](crbug.com/1225871),
+but it is pending its launch.
+
+Displaying the origin is mainly necessary for debugging purposes. Until the app
+details get launched, an alternative way to get the origin of an IWA in
+borderless mode is via the developer console with
+[`document.location.href`](https://developer.mozilla.org/en-US/docs/Web/API/Location/href).
+
+In the future when app details have been launched and origin is available, the
```

Comment:
This isn't really relevant; again, just talk about what a hypothetical OS might do in this case. Details about ChromeOS future plans can be moved to a separate doc.